elastic/kibana
# [Global Search] Limit characters for global search bar

## Summary

This PR limits the number of characters that can be input into the global search bar. The character limit can be specified with the config value `xpack.global_search_bar.input_max_limit` with a default of `1000`. When an input that exceeds the configured character limit is provided a descriptive visual notice is displayed to the user.

## Visual 
<img width="662" alt="Screenshot 2024-07-04 at 19 28 39" src="https://github.com/elastic/kibana/assets/7893459/cf30f589-fe65-40a9-b9c8-ce0f235d206e">


## How to test

- run the following command below in the browser console, which would create a string that exceeds the configured default search character limit  and copy it to your clipboard
```ts
	copy(Array.from(new Array(1001)).reduce((acc) => acc+'a', ''))
```
- open up kibana, simply paste the value that should exist in your clipboard in the global search input field and you should be presented with a result similar to the image above.
